articles/references/module_builder.md

```diff
@@ -0,0 +1,81 @@
+
+### Modeler
+If you need to fetch meta-data from a modeler that is other than `Domain` or `Services`, you need to add it here with the appropriate name so that your template can reference it. See [below](#modeler).
+
+### Type
+Consider that a given Template uses some element (or list of elements) to generate content into a file. This element Type can be specified by creating a Type that can be described in C# (or other language) namespace and type names.
+See [below](#model-type) for more info.
+
+
+## Template Settings
+
+Notice that when you click on a Template element in Intent Architect, that it will display these fields:
+
+![Template Element](images/TemplateElement.png)
+_Template Element_
+
+![Template Settings](images/TemplateSettings.png)
+_Template Settings_
+
+### Creation Mode
+The `Creation Mode` setting of a template determines what metadata format the template expects, and how files from a template must be created:
+1.	**Single File (No Model)** – create a single file without any additional metadata needed. A standalone C# class or ReadMe file would be examples of this template setting. 
+2.	**Single File (Model List)** – create a single file which takes in a list of metadata models. This is useful for creating classes which register other classes.
+3.	**File per Model** – create a file per metadata model. This is useful for creating a file based on a model such as a Domain class, Service, or DTO.
+4. **Custom** – you're on your own on this one. A minimalistic `Registration` class will be created whereby you will need to determine how many instances of that template will be created and from where.
+
+### Modeler
+The **Modeler** setting determines which _type_ of modeler the metadata should be fetched from for the template. This can be `Domain`, `Services`, `Eventing`, etc. When setting up your Module, the `Intent Module Builder` will automatically add the modeler dependency depending on which `Modeler` you have selected.
+
+### Model Type
+The **Model Type** setting specifies the object that you will be working with once the module code has been generated. You can select this Type based on `Types` that Intent Architect has been able to detect. More around this topic can be found [here](xref:IntentArchitectConcepts).
+
+### Expose Decorator Contract
+
+You might wish to create a Template that exposes an interface for a new Decorator to implement by applying the `Exposes Decorator Contract` Stereotype. The Type Fullname would be the C# Interface/class's full namespace and type name that will be used to allow Decorators to hook into this template.
+
+![Template Decorator Contract](images/TemplateDecoratorContract.png)
+
+## Decorator Settings
+
+Notice that when you click on a Decorator element in Intent Architect, that it will display these fields:
```

Comment:
```suggestion
## Decorator Stereotypes

### Decorator Settings

By default, the `Decorator Settings` and `Implements Decorator Contract` `[stereotypes](xref:Stereotypes) are added to all decorators created in the Module Builder.
```

If what I said above is true, why have two stereotypes, could the `Implements Decorator Contract` not be on the `Decorator Settings` stereotype?